### PR TITLE
Add travelperk category

### PR DESCRIPTION
**Pull request by ruuushhh**

### CODE REVIEW

File: fyle/connector.rb [hunk 1]
```diff
@@ -2559,6 +2561,33 @@
         nil
       end
     end,
+    
+    get_category_id: lambda do |connection, category_name|
+      category_id = get("#{connection["base_uri"]}/platform/v1beta/admin/categories").params(
+        'limit': 1,
+        'order': "updated_at.asc",
+        'name': "ilike.%#{category_name}%",
```
Comment (NileshPant1999) [inline fyle/connector.rb:2569]: lets not use ilike here, what is the problem with `eq`

Comment (ruuushhh) [reply fyle/connector.rb:2569]: will change it to eq

File: fyle/connector.rb [hunk 2]
```diff
@@ -2187,14 +2187,16 @@
         payload = {
           "data": input_fields["data"][0],
         }
-        category_map = { "flight" => "Airlines", "car" => "Taxi", "train" => "Train", "hotel" => "Lodging" }
+        category_map = { "flight" => "Airlines", "car" => "Taxi", "train" => "Train", "hotel" => "Lodging", "pro_v2" => "Travelperk Charges" }
 
         if category_map[input_fields["data"][0]["category_id"]]
+          system_category_id = call(:get_system_category_id, connection, category_map[input_fields["data"][0]["category_id"]])
           category_id = call(:get_category_id, connection, category_map[input_fields["data"][0]["category_id"]])
```
Comment (NileshPant1999) [inline fyle/connector.rb:2194]: let's not do it this way, let call category_id only if input_field['data'][0]['categoru_id']= 'pro_v2'
and then create the category if it is not there.

Comment (ruuushhh) [reply fyle/connector.rb:2194]: okay will add these changes

File: fyle/connector.rb [hunk 3]
```diff
@@ -2187,14 +2187,18 @@
...
-        category_map = { "flight" => "Airlines", "car" => "Taxi", "train" => "Train", "hotel" => "Lodging" }
+        category_map = { "flight" => "Airlines", "car" => "Taxi", "train" => "Train", "hotel" => "Lodging", "pro_v2" => "Travelperk Charges" }
 
         if category_map[input_fields["data"][0]["category_id"]]
-          category_id = call(:get_category_id, connection, category_map[input_fields["data"][0]["category_id"]])
-          if category_id
-            payload[:data]["category_id"] = category_id
+          system_category_id = call(:get_system_category_id, connection, category_map[input_fields["data"][0]["category_id"]])
+          category_id = nil
+          if input_fields["data"][0]["category_id"] == "pro_v2"
```
Comment (NileshPant1999) [inline fyle/connector.rb:2195]: the logic is flawed here
we need to rethink on what to do here

**Review by NileshPant1999 — APPROVED** [review]

**Pull request merged**